File: frontend/src/app/shared/model/empresa.ts
import {FornecedorPj} from './fornecedor-pj';
import {FornecedorPf} from './fornecedor-pf';

export class Empresa {

  id: number = 0;
  nomeFantasia: string;
  email: string;
  cnpj: string;
  cep: string;
  fornecedoresPJ: FornecedorPj[];
  fornecedoresPF: FornecedorPf[];

  constructor(nomeFantasia: string = '', email: string = '', cnpj: string = '', cep: string = '', fornecedoresPJ: FornecedorPj[] = [], fornecedoresPF: FornecedorPf[] = []) {
    this.nomeFantasia = nomeFantasia;
    this.email = email;
    this.cnpj = cnpj;
    this.cep = cep;
    this.fornecedoresPJ = fornecedoresPJ;
    this.fornecedoresPF = fornecedoresPF;
  }
}

File: backend/src/main/java/gldcosta/accenture/entity/Empresa.java
package gldcosta.accenture.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import gldcosta.accenture.entity.base.PessoaJuridica;
import jakarta.persistence.*;
import lombok.*;

import java.util.Set;

import static jakarta.persistence.FetchType.EAGER;

@Builder
@Getter
@Setter
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "empresa")
public class Empresa extends PessoaJuridica {

    @ManyToMany(fetch = EAGER)
    @JoinTable(
            name = "empresa_fornecedor_pf",
            joinColumns = @JoinColumn(name = "empresa_id"),
            inverseJoinColumns = @JoinColumn(name = "fornecedor_id")
    )
    private Set<FornecedorPF> fornecedoresPF;

    @ManyToMany(fetch = EAGER)
    @JoinTable(
            name = "empresa_fornecedor_pj",
            joinColumns = @JoinColumn(name = "empresa_id"),
            inverseJoinColumns = @JoinColumn(name = "fornecedor_id")
    )
    private Set<FornecedorPJ> fornecedoresPJ;
}
